.github/workflows/ci_tests_legacy.yaml

```diff
@@ -95,10 +78,18 @@ jobs:
         env:
           GH_TOKEN: ${{ github.token }}
 
+      # Install uv package manager
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+
       # Install the package that we want to test
       - name: Install the package
-        run: make install
+        run: |
+          uv run --with pip==23 --resolution lowest-direct --all-extras --dev make install
```

Comment:
Using `--resolution lowest-direct` for now, because `--resolution lowest` grabs many other transitive dependencies that may be hard to install (e.g. missing wheels for newer Python versions, so requires compilation from source). We could switch to `--resolution lowest` once the Python ecosystem does lower bound pins a bit more thoroughly (which may take years).